docs/design/hub-and-spoke/switch-validate/switch-validate.md

```diff
@@ -5,9 +5,680 @@
+      <v2GroupInfo> // without flow_id and flow_path
+    ],
+    "missing": [
+      <v2GroupInfo>
+    ],
+    "proper": [
+      <v2GroupInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "12123213",
+        "expected": {
+          <v2GroupInfo>
+        },
+        "discrepancies": {
+          <v2GroupInfo>
+        }
+      }
+    ]
+  },
+  "logical_ports": {
+    "error": "some_error",
+    "as_expected": true,
+    "excess": [
+      <v2LogicalPortInfo>
+    ],
+    "missing": [
+      <v2LogicalPortInfo>
+    ],
+    "proper": [
+      <v2LogicalPortInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "123123",
+        "expected": {
+          <v2LogicalPortInfo>
+        },
+        "discrepancies": {
+          <v2LogicalPortInfo>
+        }
+      }
+    ]
+  },
+  "meters": {
+    "as_expected": true,
+    "excess": [
+      <v2MeterInfo> // without flow_id, y_flow_id and flow_path
+    ],
+    "missing": [
+      <v2MeterInfo>
+    ],
+    "proper": [
+      <v2MeterInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "1232131",
+        "expected": {
+          <v2MeterInfo>
+        },
+        "discrepancies": {
+          <v2MeterInfo>
+        }
+      }
+    ]
+  },
+  "rules": {
+    "as_expected": true,
+    "excess": [
+      <v2RuleInfo> // without flow_id, y_flow_id and flow_path
+    ],
+    "missing": [
+      <v2RuleInfo>
+    ],
+    "proper": [
+      <v2RuleInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "asdsad",
+        "expected": {
+          <v2RuleInfo>
+        },
+        "discrepancies": {
+          <v2RuleInfo>
+        }
+      }
+    ]
+  }
+}
+```
+
+### Validate rules, meters, lags and groups installed on the switch with filters
+
+### Including specific info in response
+
+Request params
+
+To get only group info in response use query param `group`, other fields will be omitted.
+
+`GET https://{host}/v2/switches/{switch_id}/validate?include=group`
+
+Response payload
+
+```json
+{
+  "groups": {
+    "as_expected": true,
+    "excess": [
+        <v2GroupInfo> // without flow_id and flow_path
+      ],
+      "missing": [
+        <v2GroupInfo>
+      ],
+      "proper": [
+        <v2GroupInfo>
+      ],
+      "misconfigured": [
+        {
+          "id": "12123213",
+          "expected": {
+            <v2GroupInfo>
+          },
+          "discrepancies": {
+            <v2GroupInfo>
+          }
+        }
+      ]
+    }
+}
+```
+
+All other targets can be selected in same way i.e `include=meters` for meters, `include=rules` and
+so on.
+
+You can combine several flags in one query e.g. to get full info only about groups and meters use query
+params `group` and `meter`.
+
+`GET https://{host}/v2/switches/{switch_id}/validate?include=group|meter`
+
+### Excluding flow info from response
+
+Request params
+
+To exclude flow info from response use `flow_info`.
+
+`GET https://{host}/v2/switches/{switch_id}/validate?exclude=flow_info`
```

Comment:
I would like to define separate query string argument for including/excludiong flow info/reference data. So include/exclude arguments will be dedicated to select validation targets/subjects, and separate `flow_info=any_string_because_only_presence_of_the_argument_matter` (or `flow_reference`) to enable/disable adding exctra data inside result objects.